source/_integrations/zha.markdown

```diff
@@ -83,8 +83,8 @@ Some other Zigbee coordinator hardware may not support a firmware that is capabl
 - Texas Instruments based radios (via the [zigpy-znp](https://github.com/zigpy/zigpy-znp) library for zigpy)
   - [CC2652P/CC2652R/CC2652RB USB stick, module, or dev board hardware flashed with Z-Stack coordinator firmware](https://www.zigbee2mqtt.io/information/supported_adapters)
   - [CC1352P/CC1352R USB stick, module, or dev board hardware flashed with Z-Stack coordinator firmware](https://www.zigbee2mqtt.io/information/supported_adapters)
-  - [CC2538 USB stick, module, or dev board hardware flashed with Z-Stack coordinator firmware](https://www.zigbee2mqtt.io/information/supported_adapters)
-  - [CC2530/CC2531 USB stick, module, or dev board hardware flashed with Z-Stack coordinator firmware](https://www.zigbee2mqtt.io/information/supported_adapters) (not recommended for Zigbee networks with more than 20 devices)
+  - [CC2538 USB stick, module, or dev board hardware flashed with Z-Stack coordinator firmware](https://www.zigbee2mqtt.io/information/supported_adapters) (no longer recommended as only got deprecated old end-of-life firmware)
+  - [CC2530/CC2531 USB stick, module, or dev board hardware flashed with Z-Stack coordinator firmware](https://www.zigbee2mqtt.io/information/supported_adapters) (no longer recommended as uses deprecated hardware and very old end-of-life firmware, plus will not work properly at all if the whole Zigbee network has more than 15-20 devices)
```

Comment:
Should we just remove these instead?

Reply:
Probably not yet as while they are deprecated they are supported as part of legacy support in zigpy-znp. Primarily just want to deter new purchases of CC253x as heard antidotes from many new users in the community still buying them after quickly looking at the list in the ZHA docs then going to Amazon online and picking the least expensive compatible Zigbee coordinator.

I believe a lot of users already own them and can still use them fine in a very small Zigbee network, and most Zigbee devices will still work fine with a CC2530/CC2531 as coodirnator most of the time, so it is not really an issue until they need help, in which case the only solution will sometimes be to tell users to buy a newer adapters

Reply:
I think we should remove them.

Reply:
@puddly What do you think?

Reply:
This is not about what he thinks, this is about that we don't document deprecations.
Deprecated things are not listed in our documentation, with one single exception; a deprecation warning is listed when an integration as a whole is deprecated.

Since the latter is not the case, I suggest we remove it.

Reply:
Please note, this is also requiring deprecation warnings to be thrown by the integration. Has the integration be adjusted to do this?

Reply:
If so then ZHA developers would also need to consider removing all Silicon Labs EmberZNet radios that are based on Silabs older EM35x and EM35xx families of Zigbee hardware chips as Silicon Labs themselves have listed those as deprecated and obsolete for new purchase for quite a few years already, even though the latest unofficial community firmware is as of yet not missing any features or functions used by ZHA/bellows. That includes EM357 and EM358x used in Nortek GoControl QuickStick Combo Model HUSBZB-1, Bitron Video/Smabit BV AV2010/10 USB-Stick, and Telegesis ETRX357USB. Yet understand that especially HUSBZB-1 is probably still the most commonly used Zigbee adapter for ZHA as it was the very first model that the initial ZHA developer had. Again, they still work in ZHA, but they should not be recommended to beginners for new purchases.

Reply:
I'm confused, so they are not deprecated?

Reply:
Not deprecated in the zigpy-znp library code, no. The hardware is deprecated by Texas Instruments, as in they marked it as deprecated and stopped developing firmware for it a long time ago. That now lead to them now being "not recommend" by zigpy-znp.

Reply:
> I think @Hedda is trying to discourage new users from _buying_ the hardware while still documenting the fact that ZHA will _work_ with it, since people switching between the two major Zigbee integrations can do so.

Exactly! My point with this PR was about making the listing of those not sound like an advertisement or promotion. That is, I am just trying to make it more clear that it is not advised today to go out a buy a CC2530/CC2531/CC2538 hardware. That does however not stop CC253x from working or prevent new installations with it as support for it has not been removed.

Reply:
I think that the current situation with CC253x based Zigbee coordinator adapters can to a degree be somewhat compared to the Z-Wave JS wanting to post a warning in the Z-Wave JS integration docs about a specific hardware series -> https://github.com/home-assistant/home-assistant.io/pull/21230

Though for CC253x based Zigbee coordinator in ZHA is not a bug but instead just that fact the CC253x hardware is slow and underpowered and unmaintained firmware is getting old for modern use cases so want to posting a discourage warning to new potential users of ZHA that they should avoid buying Zigbee coordinator adapters based on older chips even if those are still compatible.

Reply:
> while they are deprecated they are supported as part of legacy support in zigpy-znp. Primarily just want to deter new purchases of CC253x as **keep hearing explanations and anecdotes from many new users in the community still buying them after quickly looking at the list in the ZHA docs** then going to Amazon online and picking the least expensive compatible Zigbee coordinator without doing any further research about the hardware or firmware.

A perfect new example from today -> https://community.home-assistant.io/t/using-a-zigbee-cc2531-dongle/383454/8

"_Whilst i understand that the CC2531 are not recommended, nowhere do any of the docs say they are unsupported / do not work…infact, the CC2531 adapters are listed as being [known working radios](https://www.home-assistant.io/integrations/zha#known-working-zigbee-radio-modules)
(All taken from currently published and valid documentation as far as i see)_

_I have indeed followed all the guidance and tips and best practices, but i still see really flakey connections and little reliability…to the point of none useability.
So i am starting to think that either the adapter i have is garbage, or the integrations / add-ons do not fully support CC2531, which imho, is different to not being recommended_"